manifests/agent.pp

```diff
@@ -321,20 +321,12 @@
   # but only if the variables aren't provided.
 
   if $zabbix_version != $zabbix::params::zabbix_version and $additional_service_params == $zabbix::params::additional_service_params {
-    $real_additional_service_params = versioncmp($zabbix_version, '3.0') ? {
-      1  => '--foreground',
-      0  => '--foreground',
-      -1 => '',
-    }
+    $real_additional_service_params = '--foreground'
   } else {
     $real_additional_service_params = $additional_service_params
   }
   if $zabbix_version != $zabbix::params::zabbix_version and $service_type == $zabbix::params::service_type {
-    $real_service_type = versioncmp($zabbix_version, '3.0') ? {
-      1  => 'simple',
-      0  => 'simple',
-      -1 => 'forking',
-    }
+    $real_service_type = 'simple'
   } else {
     $real_service_type = $service_type
   }
```

Comment:
I don't understand this logic. If there is no differentiation on version anymore, can't you just trust `$service_type`? Same above with `additional_service_params`.

Reply:
ah good  catch. I cleaned that up